### PR TITLE
RDoc-799 Changes API (WIP)

### PR DESCRIPTION
**Pull request by talweiss1982**
I was asked to send a PR ans switch to work on a support issue, please review and possibly merge.

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/client-api/changes/how-to-subscribe-to-document-changes.dotnet.markdown [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+### Example
+
+{CODE document_changes_2@ClientApi\Changes\HowToSubscribeToDocumentChanges.cs /}
+
+{PANEL/}
+
+{PANEL:ForDocumentsInCollection}
+
+To observe all document changes in particular collection use `ForDocumentInCollection` method. This method filters documents by `Raven-Entity-Name` metadata property value.
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/changes/how-to-subscribe-to-document-changes.dotnet.markdown]: Change `Raven-Entity-Name` to `@collection`

File: Documentation/4.0/Raven.Documentation.Pages/client-api/changes/how-to-subscribe-to-document-changes.dotnet.markdown [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+| Parameters | | |
+| ------------- | ------------- | ----- |
+| **collectionName** | string | Name of document collection for which notifications will be processed. |
+
+| Return Value | |
+| ------------- | ----- |
+| IChangesObservable<[DocumentChange](../../client-api/changes/how-to-subscribe-to-document-changes#documentchange)> | Observable that allows to add subscriptions to notifications for given document collection name. |
+
+{INFO Overload with `TEntity` type uses `Conventions.GetTypeTagName` to get collection name. /}
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/changes/how-to-subscribe-to-document-changes.dotnet.markdown]: IIRC now it is `GetCollectionName` not `GetTypeTagName`

File: Documentation/4.0/Raven.Documentation.Pages/client-api/changes/how-to-subscribe-to-document-changes.dotnet.markdown [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+To observe all document changes for given type use `ForDocumentsOfType` method. This method filters documents by `Raven-Clr-Type` metadata property value.
+
+## Syntax
+
+{CODE document_changes_6@ClientApi\Changes\HowToSubscribeToDocumentChanges.cs /}
+
+| Parameters | | |
+| ------------- | ------------- | ----- |
+| **typeName** or **type** | string or Type | Name of type or type for which notifications will be processed. If default conventions are used, the full type name without version information should be passed. See `Raven.Client.Document.ReflectionUtil.GetFullNameWithoutVersionInformation`. |
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/changes/how-to-subscribe-to-document-changes.dotnet.markdown]: Let's change the namespace to `Raven.Client.Util.ReflectionUtil.GetFullNameWithoutVersionInformation`

File: Documentation/4.0/Raven.Documentation.Pages/client-api/changes/how-to-subscribe-to-document-changes.dotnet.markdown [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+| **BulkInsertEnded** | `8` |
+| **BulkInsertError** | `16` |
+| **DeleteOnTombstoneReplication** | `32` |
+| **Conflict** | `64` |
+| **Common** | `Put & Delete` |
+{PANEL/}
+
+## Remarks
+
+{INFO To get more method overloads, especially the ones supporting delegates, please add [Reactive Extensions](http://nuget.org/packages/Rx-Main) package to your project. /}
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/changes/how-to-subscribe-to-document-changes.dotnet.markdown]: Its called `System.Reactive` now https://www.nuget.org/packages/System.Reactive

File: Documentation/4.0/Raven.Documentation.Pages/client-api/changes/what-is-changes-api.dotnet.markdown [hunk 5]
```diff
@@ -1 +1,51 @@
...
+
+## Unsubscribing
+
+In order to end subscription (stop listening for particular notifications) you must `Dispose` it.
+
+{CODE changes_2@ClientApi\Changes\WhatIsChangesApi.cs /}
+
+## Remarks
+
+{INFO To get more method overloads, especially the ones supporting delegates, please add [Reactive Extensions](http://nuget.org/packages/Rx-Main) package to your project. /}
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/changes/what-is-changes-api.dotnet.markdown]: Its called `System.Reactive` now https://www.nuget.org/packages/System.Reactive

File: Documentation/4.0/Raven.Documentation.Pages/client-api/changes/how-to-subscribe-to-document-changes.dotnet.markdown (hunk 4 above)
Comment (ppekrol) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/changes/how-to-subscribe-to-document-changes.dotnet.markdown]: But maybe we need just `System.Reactive.Core` or `Linq` can you check that?

Comment (talweiss1982) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/changes/how-to-subscribe-to-document-changes.dotnet.markdown]: this PR already contains System.Reactive.Core for the sample code and its enough

**Pull request merged**